tests/h/traversal/annotation_test.py

```diff
@@ -1,348 +1,134 @@
-    # transferred over to the annotation as expected
-    # As such, this has sort of a partial and wonky re-implementation of
-    # ``h.models.Group.__acl__``
-    # This is a symptom of the disease that is splitting ACL concerns between
-    # traversal/resources and model classes
-    # TODO: Refactor once we're able to move ACLs off of models
-    def __init__(self, principals):
-        acl = []
-        for p in principals:
-            acl.append((security.Allow, p, Permission.Group.READ))
-            if p == security.Everyone:
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.FLAG)
-                )
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.MODERATE)
-                )
-            else:
-                acl.append((security.Allow, p, Permission.Group.FLAG))
-                # Normally, the ``moderate`` permission would only be applied
-                # to the admin (creator) of a group, but this ``FakeGroup``
-                # is indeed fake. Tests in this module are merely around whether
-                # this permission is translated appropriately from a group
-                # to an annotation context (i.e. it should not be applied
-                # to private annotations)
-                acl.append((security.Allow, p, Permission.Group.MODERATE))
-        self.__acl__ = acl
-
-
-@pytest.mark.usefixtures("groupfinder_service", "links_service")
 class TestAnnotationRoot:
-    def test_it_does_not_assign_create_permission_without_authenticated_user(
-        self, set_permissions, pyramid_request
-    ):
-        set_permissions()
-
-        context = AnnotationRoot(pyramid_request)
+    def test_create_permission_requires_authenticated_user(self, root, permits):
+        assert permits(root, [security.Authenticated], Permission.Annotation.CREATE)
+        assert not permits(root, [], Permission.Annotation.CREATE)
 
-        assert not pyramid_request.has_permission(Permission.Annotation.CREATE, context)
-
-    def test_it_assigns_create_permission_to_authenticated_request(
-        self, set_permissions, pyramid_request
+    def test_annotation_lookup(
+        self,
+        root,
+        pyramid_request,
+        AnnotationContext,
+        storage,
+        groupfinder_service,
+        links_service,
     ):
-        set_permissions(
-            "acct:adminuser@foo", principals=[pyramid.security.Authenticated]
+        context = root[sentinel.annotation_id]
+
+        assert context == AnnotationContext.return_value
+        storage.fetch_annotation.assert_called_once_with(
+            pyramid_request.db, sentinel.annotation_id
+        )
+        AnnotationContext.assert_called_once_with(
+            storage.fetch_annotation.return_value, groupfinder_service, links_service
         )
 
-        context = AnnotationRoot(pyramid_request)
+    def test_failing_annotation_lookup(self, root, storage):
```

Comment:
Maybe something nice and concrete and explicit like:

```suggestion
    def test_it_raises_KeyError_if_the_annotation_id_isnt_in_the_DB(self, root, storage):
```

Or just `_it_raises_` instead of `_it_raises_KeyError_`

Reply:
Yep I've renamed these.